lualib/resty/kong/request.lua

```diff
@@ -0,0 +1,35 @@
+local ffi = require "ffi"
+local base = require "resty.core.base"
+local get_request = base.get_request
+
+local C = ffi.C
+local ffi_new = ffi.new
+local ffi_str = ffi.string
+
+local subsystem = ngx.config.subsystem
+
+local get_header
+if subsystem == "http" then
+    ffi.cdef[[
+        ngx_str_t *ngx_http_lua_kong_request_get_header(
+            ngx_http_request_t *r, ngx_str_t name, size_t search_limit);
+    ]]
+
+    local DEFAULT_HEADER_LIMIT = 100
+
+    local name_ffi = ffi_new("ngx_str_t")
+    get_header = function (name, limit)
+        local r = get_request()
+        name_ffi.data = name
```

Comment:
I think `name_ffi` should be changed to a simple `name_str`. `ffi` is a bit verbose.